src/Crypto/WebAuthn/AttestationStatementFormat/TPM.hs

```diff
@@ -149,33 +149,53 @@ data TPMSAttest = TPMSAttest
 -- section 8.3
 type TPMAObject = Word32
 
+-- | The TPMS_RSA_PARMS structure as specified in
+-- [TPMv2-Part2](https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf)
+-- section 12.2.3.5.
+data TPMSRSAParms = TPMSRSAParms
+  { tpmsrpSymmetric :: Word16,
+    tpmsrpScheme :: Word16,
+    tpmsrpKeyBits :: Word16,
+    tpmsrpExponent :: Word32
+  }
+  deriving (Eq, Show, Generic, ToJSON)
+
+-- | The TPMS_ECC_PARMS structure as specified in
+-- [TPMv2-Part2](https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf)
+-- section 12.2.3.6.
+data TPMSECCParms = TPMSECCParms
+  { tpmsepSymmetric :: Word16,
+    tpmsepScheme :: Word16,
+    tpmsepCurveId :: Cose.CoseCurveECDSA,
+    tpmsepkdf :: Word16
+  }
+  deriving (Eq, Show, Generic, ToJSON)
+
 -- | The TPMU_PUBLIC_PARMS structure as specified in
 -- [TPMv2-Part2](https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf)
 -- section 12.2.3.7.
 data TPMUPublicParms
-  = TPMSRSAParms
-      { tpmsrpSymmetric :: Word16,
-        tpmsrpScheme :: Word16,
-        tpmsrpKeyBits :: Word16,
-        tpmsrpExponent :: Word32
-      }
-  | TPMSECCParms
-      { tpmsepSymmetric :: Word16,
-        tpmsepScheme :: Word16,
-        tpmsepCurveId :: Cose.CoseCurveECDSA,
-        tpmsepkdf :: Word16
-      }
+  = TPMUPublicParmsRSA TPMSRSAParms
+  | TPMUPublicParmsECC TPMSECCParms
```

Comment:
Are you guys okay with this naming convention for the constructors? I'm open to making any changes.